src/lib/sql/sql_pipeline_statement.cpp

```diff
@@ -175,24 +175,31 @@ const std::shared_ptr<SQLQueryPlan>& SQLPipelineStatement::get_query_plan() {
 
   const auto* statement = get_parsed_sql_statement()->getStatement(0);
 
+  auto check_same_mvcc_mode = [this](const SQLQueryPlan& plan) {
+    if (plan.tree_roots().front()->transaction_context_is_set()) {
+      Assert(_use_mvcc == UseMvcc::Yes, "Trying to use MVCC cached query without a transaction context.");
+    } else {
+      Assert(_use_mvcc == UseMvcc::No, "Trying to use non-MVCC cached query with a transaction context.");
+    }
+  };
+
   try {
     if (const auto cached_plan = SQLQueryCache<SQLQueryPlan>::get().try_get(_sql_string)) {
       // Handle query plan if statement has been cached
       auto& plan = *cached_plan;
 
       DebugAssert(!plan.tree_roots().empty(), "QueryPlan retrieved from cache is empty.");
-      if (plan.tree_roots().front()->transaction_context_is_set()) {
-        Assert(_use_mvcc == UseMvcc::Yes, "Trying to use MVCC cached query without a transaction context.");
-      } else {
-        Assert(_use_mvcc == UseMvcc::No, "Trying to use non-MVCC cached query with a transaction context.");
-      }
+      check_same_mvcc_mode(plan);
```

Comment:
`assert_mvcc_mode_matches`?

Reply:
Also, should this be public? If not, we need an underscore.

Reply:
This is a simple lamdba that is defined within the function, so it's not public. But naming can be changed.

Reply:
oups, you're right